astropy/coordinates/builtin_frames/itrs_to_observed_transforms.py

```diff
@@ -0,0 +1,146 @@
+    mat = (minus_y
+           @ rotation_matrix(elong, 'z', unit=u.radian))
+    return mat
+
+
+def altaz_to_hadec_mat(lat):
+    # form AltAz to HADec matrix
+    elat = lat.to_value(u.radian)
+    z180 = np.eye(3)
+    z180[0][0] = -1.0
+    z180[1][1] = -1.0
+    mat = (z180
+           @ rotation_matrix(PIOVER2 - elat, 'y', unit=u.radian))
+    return mat
+
+
+def add_refraction(aa_crepr, observed_frame):
+    # add refraction to AltAz cartesian representation
+    refa, refb = erfa.refco(
+        observed_frame.pressure.to_value(u.hPa),
+        observed_frame.temperature.to_value(u.deg_C),
+        observed_frame.relative_humidity.value,
+        observed_frame.obswl.to_value(u.micron)
+    )
+    #reference: erfa.atioq()
+    norm, uv = erfa.pn(aa_crepr.get_xyz(xyz_axis=-1).to_value())
+    # Cosine and sine of altitude, with precautions.
+    sel = np.maximum(uv[..., 2], SELMIN)
+    cel = np.maximum(np.sqrt(uv[..., 0] ** 2 + uv[..., 1] ** 2), CELMIN)
+    # A*tan(z)+B*tan^3(z) model, with Newton-Raphson correction.
+    tan_z = cel / sel
+    w = refb * tan_z ** 2
+    delta_el = (refa + w) * tan_z / (1.0 + (refa + 3.0 * w) / (sel ** 2))
+    # Apply the change, giving observed vector
+    cosdel = 1.0 - 0.5 * delta_el ** 2
+    f = cosdel - delta_el * sel / cel
+    uv[..., 0] *= f
+    uv[..., 1] *= f
+    uv[..., 2] = cosdel * uv[..., 2] + delta_el * cel
+    # Need to renormalize to get agreement with CIRS->Observed on distance
+    norm2, uv = erfa.pn(uv)
+    uv = erfa.sxp(norm, uv)
+    return CartesianRepresentation(uv, xyz_axis=-1, unit=aa_crepr.x.unit, copy=False)
+
+
+def remove_refraction(aa_crepr, observed_frame):
+    # remove refraction from AltAz cartesian representation
+    refa, refb = erfa.refco(
+        observed_frame.pressure.to_value(u.hPa),
+        observed_frame.temperature.to_value(u.deg_C),
+        observed_frame.relative_humidity.value,
+        observed_frame.obswl.to_value(u.micron)
+    )
+    # reference: erfa.atoiq()
+    norm, uv = erfa.pn(aa_crepr.get_xyz(xyz_axis=-1).to_value())
+    # Cosine and sine of altitude, with precautions.
+    sel = np.maximum(uv[..., 2], SELMIN)
+    cel = np.sqrt(uv[..., 0] ** 2 + uv[..., 1] ** 2)
+    # A*tan(z)+B*tan^3(z) model
+    tan_z = cel / sel
+    delta_el = (refa + refb * tan_z ** 2) * tan_z
+    # Apply the change, giving observed vector.
+    az, el = erfa.c2s(uv)
+    el -= delta_el
+    uv = erfa.s2c(az, el)
+    uv = erfa.sxp(norm, uv)
+    return CartesianRepresentation(uv, xyz_axis=-1, unit=aa_crepr.x.unit, copy=False)
+
+
+@frame_transform_graph.transform(FunctionTransformWithFiniteDifference, ITRS, AltAz)
+@frame_transform_graph.transform(FunctionTransformWithFiniteDifference, ITRS, HADec)
+def itrs_to_observed(itrs_coo, observed_frame):
+    if (np.any(itrs_coo.location != observed_frame.location) or
+            np.any(itrs_coo.obstime != observed_frame.obstime)):
+        itrs_coo = itrs_coo.transform_to(ITRS(obstime=observed_frame.obstime,
```

Comment:
I thought that I had gotten around that by making no change in the current behavior for that situation, which includes no warning. But, sure I can include a new warning for that situation. Do you have a suggestion as to what that warning should be? 

Perhaps, I should also include a warning if the obstimes are the same but the locations are different. Though less dire, the consequence would be to add the difference in stellar aberration between the points of view of the two observers.

Reply:
Also note that, as currently written, I could drop my ITRS<->Observed transforms completely and there would be no change in behavior. Topocentric ITRS to Observed transforms would just go through the CIRS without any warnings. Doing this would just be 4 times slower is all.